assets/src/scripts/models.js

```diff
@@ -42,33 +42,126 @@ export function getTimeseries({sites, params=['00060'], startDate=null, endDate=
-                                    date)}\n${value.value} ${series.variable.unit.unitCode}`
-                            };
-                        })
-                    };
+                let data = JSON.parse(response);
+                return data.value.timeSeries.map(series => {
+                        let startDate = new Date(series.values[0].value[0].dateTime);
+                        let endDate = new Date(
+                            series.values[0].value.slice(-1)[0].dateTime);
+                        return {
+                            code: series.variable.variableCode[0].value,
+                            variableName: series.variable.variableName,
+                            variableDescription: series.variable.variableDescription,
+                            seriesStartDate: startDate,
+                            seriesEndDate: endDate,
+                            values: series.values[0].value.map(value => {
+                                let date = new Date(value.dateTime);
+                                return {
+                                    time: date,
+                                    value: parseFloat(value.value),
+                                    label: `${formatTime(
+                                        date)}\n${value.value} ${series.variable.unit.unitCode}`
+                                };
+                            })
+                        };
+                    }
+                );
+            },
+            (error) => {
+                return error;
+            });
+}
+export function getSiteStatistics({sites, statType, params=['00060']}) {
+    let url = `${SERVICE_ROOT}/stat/?format=rdb&sites=${sites.join(',')}&statReportType=daily&statTypeCd=${statType}&parameterCd=${params.join(',')}`;
+    return get(url);
+}
+
+
+/**
+ * Function to parse RDB to Objects
+ */
+export function parseRDB(rdbData) {
+    let rdbLines = rdbData.split('\n');
+    let dataLines = rdbLines.filter(rdbLine => rdbLine[0] !== '#').filter(rdbLine => rdbLine.length > 0);
+    // remove the useless row
+    dataLines.splice(1, 1);
+    let recordData = [];
+    if (dataLines.length > 0) {
+        let headers = dataLines.shift().split('\t');
+        for (let dataLine of dataLines) {
+            let data = dataLine.split('\t');
+            let dataObject = {};
+            for (let i=0; i < headers.length; i++) {
+                dataObject[headers[i]] = data[i];
+            }
+            recordData.push(dataObject);
+        }
+    }
+    return recordData;
+}
+
+/**
+ * Determine if a given year is a leap year
+ *
+ * @param year
+ * @returns {boolean}
+ */
+export function isLeapYear(year) {
+    return (year % 4) === 0;
+}
+
+/**
+ *  Read median RDB data into something that makes sense
+ *
+ * @param medianData
+ * @param timeSeries
+ * @params subsetDays
+ * @returns {Array}
+ */
+export function parseMedianData(medianData, timeSeries) {
```

Comment:
I wonder if this function would be easier to understand if, rather than taking in `timeSeries`, it took in only the derived values that are needed - like start and end date?